### PR TITLE
refactor templates

### LINKED ISSUE #703: Surveys/polls/quizzes appear at top of Section regardless of Section items order
Right now s/p/q appear at the top of Section Pages no matter where they appear in Section order:

Website:
image

Admin Panel:
image

Urgent and simple fix:

- [ ] Move block of s/p/q content to bottom of list on user-facing site (to mirror V1)

Full fix:

- [ ] s/p/q appear in order (individually, not as a block) among the other items in the section (also referenced in #702)

### LINKED ISSUE #758: Margins for article content should be reduced
By comparison nytimes.com, yahoo.com and many other websites are much thinner on the margins. I see this on mobile, obviously it's less obvious on desktop and tablet with the wider screen.

Was there a conscious decision behind having the wide margins?

### LINKED ISSUE #770: Show top-level section titles for featured articles
#685 asked for an option to display section titles for featured articles.

Our implementation shows the title of the **parent** section.
image
After seeking clarification, we should display the title of the **top-level** section that the article is in (e.g. Youth, Covid-19 for the global site, see screenshot in #685).
If possible, also inherit the text color of that section for displaying the text.

### PR DESCRIPTION
**Pull request by sheralim012**
fixes #703 
fixes #770 
fixes #758 

- removed one questionnaire rendering
- removed unused article chooser block
- simplified article block
- removed unused footer in homepage context
- removed unused featured content in homepage context
- section children are rendered in the order specified in admin
- removed unused breadcrumbs from article context
- only live recommended articles are shown in article
- on homepage featured content show top level section
- unified article card rendering
- unified section card rendering
- unified questionnaires card rendering
- fixes featured content in section rendering
- removed unused template tags from home app
- removed unused template tags from questionnaires app
- unified search result rendering
- always send pages in their specific form to template

### CODE REVIEW

File: home/templates/home/tags/banners_list.html [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
+            {% if page.banner_link_page %}
+                <a href="{% pageurl page.banner_link_page %}">
+            {% elif page.external_link %}
+                <a href="{{ page.external_link }}" target="_blank">
             {% else %}
                 <a href="#">
             {% endif %}
-                <img src="{{ banner_img.url }}" alt="{{ banner_img.alt }}" >
+            {% image page.banner_image original as banner_img %}
+            <img src="{{ banner_img.url }}" alt="{{ banner_img.alt }}">
```
Comment (ehmadzubair) [inline home/templates/home/tags/banners_list.html:14]: maybe remove it

File: home/templates/home/tags/featured_content_card.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{% load static wagtailcore_tags wagtailimages_tags %}
+
+<svg width="18" height="16" viewBox="0 0 18 16" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (ehmadzubair) [inline home/templates/home/tags/featured_content_card.html:3]: Any idea why this is hardcoded?

File: home/templates/home/tags/featured_content_card.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+          fill="#303030"/>
+</svg>
+<a href="{% pageurl page %}">{{ page.title }}</a>
+{% include 'home/tags/right_angle_bracket.html' %}
```
Comment (ehmadzubair) [inline home/templates/home/tags/featured_content_card.html:9]: Make this a templatetag

**Review by ehmadzubair — CHANGES_REQUESTED** [review]

**Comment by geoo89** [comment]
- The `LARGER IMAGE FOR TOP PAGE IN LIST AS IN V1` flag is not respected: In sections, the top article is always displayed with a larger image.
- PageLinkPages don't show in sections
- Questionnaire links don't use a section's color scheme
    - Subsections: Should they use the section's color scheme as well? Only implement if trivial. We might have to check back with Chris and revisit the coloring schemes as having everything in the section color scheme might be too intrusive.
- Adding featured content crashes a section:

```
Environment:


Request Method: GET
Request URL: http://localhost:8000/en/sections/covid-19/

Django Version: 3.1.11
Python Version: 3.6.9
Installed Applications:
['home',
 'search',
 'iogt_users',
 'comments',
 'iogt_content_migration',
 'questionnaires',
 'messaging',
 'django.contrib.humanize',
 'wagtail_localize',
 'wagtail_localize.locales',
 'wagtail.contrib.forms',
 'wagtail.contrib.redirects',
 'wagtail.embeds',
 'wagtail.sites',
 'wagtail.users',
 'wagtail.snippets',
 'wagtail.documents',
 'wagtail.images',
 'wagtail.search',
 'wagtail.admin',
 'wagtail.core',
 'wagtail.contrib.modeladmin',
 'wagtailmenus',
 'wagtailmedia',
 'wagtailmarkdown',
 'wagtail_transfer',
 'wagtailsvg',
 'wagtail.contrib.settings',
 'django_comments_xtd',
 'django_comments',
 'modelcluster',
 'taggit',
 'allauth',
 'allauth.account',
 'allauth.socialaccount',
 'rest_framework',
 'sass_processor',
 'translation_manager',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.sites',
 'django_extensions']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'wagtail.contrib.redirects.middleware.RedirectMiddleware',
 'iogt_users.middlewares.RegistrationSurveyRedirectMiddleware',
 'external_links.middleware.RewriteExternalLinksMiddleware',
 'iogt.middleware.CacheControlMiddleware']


Template error:
In template /data/idems/iogt/iogt/home/templates/home/tags/featured_content_card.html, error at line 4
   21
   1 : {% load image_tags %}
   2 : {% load static wagtailcore_tags wagtailimages_tags %}
   3 : 
   4 :  {% render_png_from_svg page.get_icon_url width=16 height=16 %} 
   5 : <a href="{% pageurl page %}">{{ page.title }}</a>
   6 : {% include 'home/tags/angle_bracket.html' %}

Traceback (most recent call last):
  File "/data/idems/iogt/iogt/home/models.py", line 1010, in get_png_image
    obj = cls.objects.get(svg_path=svg_path, fill_color=fill_color, stroke_color=stroke_color)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/db/models/query.py", line 431, in get
    self.model._meta.object_name

During handling of the above exception (SVGToPNGMap matching query does not exist.), another exception occurred:
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/core/handlers/base.py", line 204, in _get_response
    response = response.render()
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/response.py", line 105, in render
    self.content = self.rendered_content
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/response.py", line 83, in rendered_content
    return template.render(context, self._request)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 170, in render
    return self._render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/loader_tags.py", line 192, in render
    return template.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 172, in render
    return self._render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/defaulttags.py", line 211, in render
    nodelist.append(node.render_annotated(context))
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/loader_tags.py", line 192, in render
    return template.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 172, in render
    return self._render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv/lib/python3.6/site-packages/django/template/library.py", line 214, in render
    _dict = self.func(*resolved_args, **resolved_kwargs)
  File "/data/idems/iogt/iogt/home/templatetags/image_tags.py", line 13, in render_png_from_svg
    'image_url': SVGToPNGMap.get_png_image(absolute_path, fill_color=fill_color, stroke_color=stroke_color).url,
  File "/data/idems/iogt/iogt/home/models.py", line 1012, in get_png_image
    png_image = convert_svg_to_png_bytes(svg_path, fill_color=fill_color, stroke_color=stroke_color, width=32)
  File "/data/idems/iogt/iogt/home/utils/image.py", line 10, in convert_svg_to_png_bytes
    with open(svg_file_path, 'r', encoding='utf-8') as f:

Exception Type: IsADirectoryError at /en/sections/covid-19/
Exception Value: [Errno 21] Is a directory: '/data/idems/iogt/iogt'
```

**Comment by sheralim012** [comment]
@geoo89 changes addressed

**Comment by geoo89** [comment]
Index page description isn't displayed for articles.

**Comment by sheralim012** [comment]
pushed a fix for index page description for articles

**Comment by geoo89** [comment]
The section color and background color should not affect the text/background of the elements of a section.

More sophisticated coloring rules to be determined later.

**Comment by sheralim012** [comment]
@geoo89 changes addressed

**Pull request merged**